src/pattern.rs

```diff
@@ -14,41 +14,23 @@
 //! * `%t` - The target of the log message.
 //!
 
-use std::borrow::ToOwned;
+//mod parse_pattern;
+
 use std::default::Default;
 use std::error;
 use std::fmt;
 use std::thread;
 use std::io;
 use std::io::Write;
+use parser::{TimeFmt, Chunk, parse_pattern};
+use nom;
 
 use log::{LogRecord, LogLevel};
 use time;
 
-#[derive(Debug)]
-#[cfg_attr(test, derive(PartialEq))]
-enum TimeFmt {
-    Rfc3339,
-    Str(String),
-}
-
-#[derive(Debug)]
-#[cfg_attr(test, derive(PartialEq))]
-enum Chunk {
-    Text(String),
-    Time(TimeFmt),
-    Level,
-    Message,
-    Module,
-    File,
-    Line,
-    Thread,
-    Target,
-}
-
 /// An error parsing a `PatternLayout` pattern.
 #[derive(Debug)]
-pub struct Error(String);
+pub struct Error(pub String);
```

Comment:
Can this get reverted? You can use a private trait at the crate root to emulate crate-local privacy. e.g.
```rust
trait ErrorInternals {
    fn new(message: String) -> Error;
}
```